src/socket-to-conn.ts

```diff
@@ -49,11 +63,11 @@ module.exports = (socket, options = {}) => {
       : undefined,
 
     // If the remote address was passed, use it - it may have the peer ID encapsulated
-    remoteAddr: options.remoteAddr,
+    remoteAddr: options.remoteAddr ?? toMultiaddr(socket.remoteAddress ?? '', socket.remotePort ?? ''),
```

Comment:
I had to add this conversion because `remoteAddr` is required in `MultiaddrConnection` interface.

Reply:
I made `remoteAddr` option mandatory because all usages of `ToConnectionOptions` specify it. Please verify.